lib/excerpt_parser.rb

```diff
@@ -144,11 +144,19 @@ def end_element(name)
     when "details"
       @in_details_depth -= 1
       if @in_details_depth == 0
-        full = "<details><summary>#{clean(@summary_contents)}</summary>#{clean(@detail_contents)}</details>"
-        if @current_length + full.length > @length
-          @excerpt << "<details class='disabled'><summary>#{@summary_contents[0..@length]}</summary></details>"
+        @summary_contents = clean(@summary_contents)
+        @detail_contents = clean(@detail_contents)
+
+        if @current_length + @summary_contents.length >= @length
+          characters(@summary_contents, true, true, false,
```

Comment:
These `true, true, false` arguments should be made named parameters. Otherwise it gets hard to understand.